### PR TITLE
feat(aws-lambda-stepfunctions): aws-lambda-stepfunctions implementation

### PR DESCRIPTION
**Pull request by mickychetta**
*Issue #271 , if available:*

*Description of changes:*
Transitioning from aws-lambda-step-function to new construct aws-lambda-stepfunctions to adhere to consistent naming conventions. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 513cc143e1f744c67c6dbba2832b292dc0129c2c
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=65d9af3b-9009-4ced-a160-01a299c1b376%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-lambda-step-function/lib/index.ts [hunk 1]
```diff
@@ -20,9 +20,9 @@ import { Construct } from '@aws-cdk/core';
 import * as logs from '@aws-cdk/aws-logs';
 
 /**
- * @summary Properties for the LambdaToStepFunction class.
+ * @summary Properties for the LambdaToStepFunctions class.
  */
-export interface LambdaToStepFunctionProps {
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-step-function/lib/index.ts:25]: We don't want to change anything in the old interface, if we change the name of the props interface and the construct class we will break any existing customer code.

File: source/patterns/@aws-solutions-constructs/aws-lambda-step-function/lib/index.ts [hunk 2]
```diff
@@ -62,23 +62,23 @@ export interface LambdaToStepFunctionProps {
...
    * @param {string} id - this is a a scope-unique id.
-   * @param {LambdaToStepFunctionProps} props - user provided props for the construct.
+   * @param {LambdaToStepFunctionsProps} props - user provided props for the construct.
    * @since 0.8.0
    * @access public
    */
-  constructor(scope: Construct, id: string, props: LambdaToStepFunctionProps) {
+  constructor(scope: Construct, id: string, props: LambdaToStepFunctionsProps) {
     super(scope, id);
     defaults.CheckProps(props);
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-step-function/lib/index.ts:83]: This code in the constructor should all be deleted. A new LambdaToStepfunctionsProps object should be created and populated from the incoming LambdaToStepFunctionProps values. Then the constructor will make a call like:

const wrappedConstruct = new LambdaToStepfunctions(this,  `${id}-wrapped`, newLambdaToStepfunctionsProps);

// After that, the exposed properties of the construct class will need to be populated from the exposed properties of wrapptedConstruct.

File: source/patterns/@aws-solutions-constructs/aws-lambda-stepfunctions/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_lambda_stepfunctions`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-lambda-stepfunctions`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.lambdastepfunctions`|
+
+This AWS Solutions Construct implements an AWS Lambda function connected to an AWS Step Functions.
+
+Here is a minimal deployable pattern definition in Typescript:
+
+``` typescript
+import { LambdaToStepFunctions } from '@aws-solutions-constructs/aws-lambda-stepfunctions';
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-stepfunctions/README.md:30]: Name needs to be LambdaToStepfunctions to match design guidelines (we only uppercase the first letter - while that seems odd here if you think of LambdaToSqs the reason for the pattern becomes clearer). 

(same for Props interface)

**Review by biffgaut — CHANGES_REQUESTED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 5db262609ae86d151f0657732608121a9f13af00
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f79e10e3-52b7-44a5-bbdc-ec14fbb1929f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: b3ab8e4cca6dc0d88dd2a0416b33547f6c50c94e
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7fe5c814-09e5-4cb4-93db-02adf6b6ecb3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: b87238f61cae066060db425e64b7793a8c440db8
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4d6d468b-6c3e-488e-a795-c9c420f5c229%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: ae3934c97682995977064a01f930f7f03060125f
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2b12adac-d7ea-49b0-a100-55928bf425fb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 5cee402d0419f615794a35b1f74c29926eceddda
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1d9db33a-cad1-4eae-bc7a-d4a82e7300a2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-lambda-step-function/README.md [hunk 4]
```diff
@@ -3,7 +3,9 @@
 
 ---
 
-![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+![Stability: Deprecated](https://img.shields.io/badge/STABILITY-DEPRECATED-red?style=for-the-badge)
+
+> Construct will be deprecated in favor of aws-lambda-stepfunctions in v2.x, but will be supported in v1.x as long as v1.x is supported.
```
[the remark this answers is not in the document]
Comment (biffgaut) [reply source/patterns/@aws-solutions-constructs/aws-lambda-step-function/README.md:8]: Let's replace line 8 with:

Some of our early constructs don’t meet the naming standards that evolved for the library. We are releasing completely feature compatible versions with corrected names. The underlying implementation code is the same regardless of whether you deploy the construct using the old or new name. We will support both names for all 1.x releases, but in 2.x we will only publish the correctly named constructs.

**Review by biffgaut — CHANGES_REQUESTED** [review]

**Review by biffgaut — APPROVED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 477a86adf045160d0b5381c7c502a764723748f7
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b1f33890-d369-4963-8148-6d652b3f1064%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**